jobs/build/ocp4/build.groovy

```diff
@@ -87,6 +88,37 @@ def initialize() {
     return planBuilds()
 }
 
+def setBuildType() {
+    if (!isTriggeredByOrganicLifeForms()) {
+        echo "Assuming *PROD* build, triggered by automation"
+        scratch = false
+        return
+    }
+    askBuildType()
+}
+
+def askBuildType() {
+    commonlib.inputRequired(slacklib.to(params.BUILD_VERSION)) {
+        def res = input(
+            message: 'What is the purpose of this build?',
+            parameters: [
+                [$class: 'hudson.model.ChoiceParameterDefinition',
+                 choices: 'TEST\nPROD',
+                 description: 'TEST (from SCRATCH, not pushed to quay). PROD (same as automation does)',
+                 name: 'type']
+            ]
+        )
+        scratch = res == 'TEST'
+    }
+}
+
+@NonCPS
+def isTriggeredByOrganicLifeForms() {
+    currentBuild.rawBuild.getCauses().collect {
+        it.getClass().getCanonicalName().tokenize('.').last()
+    }.contains('UserIdCause')
```

Comment:
buildlib.isBuildPermitted just checks that there's a real user that kicked off the request. i don't know which is the "best" way but it seems like they ought to work the same...